src/ContactPeople/ContactPerson/ContactPerson.js

```diff
@@ -14,11 +15,12 @@ import ContactPersonEmails from './ContactPersonEmails';
 import ContactPersonURLs from './ContactPersonURLs';
 
 const ContactPerson = ({ contact, categories, withCollapsing }) => {
-  const addresses = hydrateAddresses(categories, contact.addresses);
-  const emails = mixCategories(categories, contact.emails);
-  const phoneNumbers = mixCategories(categories, contact.phoneNumbers);
-  const urls = mixCategories(categories, contact.urls);
-  const contactCategories = transformCategoryIdsToLables(categories, contact.categories);
+  const intl = useIntl();
```

Comment:
maybe it makes sense to create hook for translated category instead of passing down intl everywhere?

Reply:
but we need to call hooks at the top level of the functional component avoiding calliing inside loops, conditions, or nested functions. We get translated categories inside other functions (`mixCategories`, `hydrateAddresses` ). Also, translation category function is called inside of `MCL` formatter.

Reply:
> but we need to call hooks at the top level of the functional component avoiding calliing inside loops, conditions, or nested functions.

very right, it would be a problem if you do hooks another way.

> We get translated categories inside other functions (mixCategories, hydrateAddresses )

and that what I'm talking about - to move them into hooks since they will need context to work

> Also, translation category function is called inside MCL formatter.

What is the problem here?

Reply:
created hook